operator/duties/controller.go

```diff
@@ -239,6 +243,41 @@ func (dc *dutyController) handleSlot(slot phase0.Slot) {
 		go dc.onDuty(&duties[i])
 	}
 
+	dc.handleSyncCommittee(slot, syncPeriod)
+	dc.handleValidatorRegistration(slot)
+}
+
+func (dc *dutyController) handleValidatorRegistration(slot phase0.Slot) {
+	// push if first time or every 10 epoch at first slot
+	epoch := dc.ethNetwork.EstimatedEpochAtSlot(slot)
+	firstSlot := dc.ethNetwork.GetEpochFirstSlot(epoch)
+	if dc.registered && (slot != firstSlot || uint64(epoch)%validatorRegistrationEpochInterval != 0) {
+		return
+	}
+	dc.registered = true
```

Comment:
is this safe to set at this point? like if the next section errors, registered will still be true, is that okay?

Reply:
removed